server/src/main/java/org/elasticsearch/rest/action/RestFieldCapabilitiesAction.java

```diff
@@ -67,7 +67,10 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
             }
             fieldRequest.fields(Strings.splitStringByCommaToArray(request.param("fields")));
         }
-        return channel -> client.fieldCaps(fieldRequest, new RestChunkedToXContentListener<>(channel));
+        return channel -> {
+            RestCancellableNodeClient cancelClient = new RestCancellableNodeClient(client, request.getHttpChannel());
+            cancelClient.fieldCaps(fieldRequest, new RestChunkedToXContentListener<>(channel));
```

Comment:
I am not convinced about this part of the change. I agree that the internal task should be made cancellable, but to make it cancellable on connection close means to me that we are acknowledging that field_caps is a potential long-running request, like search can be at times, yet the two have completely different characteristics. I think that we should rather improve field_caps performance further if needed, and while search may take a long time in certain circumstances, anytime that field_caps takes too long should be treated as a bug.

Last but not least: even if we make this part of the change, we don't know if this code is exercised: does Kibana close connections or does it need additional logic to do so? While it is clear to me that for _search, there was a problem around searches timing out on the client, with users running the same request over and over while the previous ones kept on running on the server side, I am not sure that the situation is the same with field_caps, hence I would want to verify that we are heading in the right direction before making this change.

Reply:
I am convinced now that we are good with this, no matter if Kibana exercises it or not, it is a good change in general. as cancelling is the right thing to do when a connection is closed. I will mark this comment resolved.